packages/template-retail-react-app/app/components/store-locator-modal/store-locator-content.jsx

```diff
@@ -58,17 +165,66 @@ const StoreLocatorContent = ({form, submitForm, storesInfo, searchStoresParams})
+                            : searchStoresParams.postalCode !== undefined
+                            ? `${intl.formatMessage(
                                   {
-                                      id: 'store_locator.description.viewing_within',
-                                      defaultMessage: 'Viewing stores within {distance} miles'
+                                      id: 'store_locator.description.viewing_near_postal_code',
+                                      defaultMessage:
+                                          'Viewing stores within {distance}{distanceUnit} of {postalCode} in '
                                   },
-                                  {distance: STORE_LOCATOR_DISTANCE}
+                                  {
+                                      distance: STORE_LOCATOR_DISTANCE,
+                                      distanceUnit: STORE_LOCATOR_DISTANCE_UNIT,
+                                      postalCode: searchStoresParams.postalCode
+                                  }
                               )}
+                            ${
+                                SUPPORTED_STORE_LOCATOR_COUNTRIES.length !== 0
+                                    ? intl.formatMessage(
+                                          SUPPORTED_STORE_LOCATOR_COUNTRIES.find(
+                                              (o) =>
+                                                  o.countryCode === searchStoresParams.countryCode
+                                          ).countryName
+                                      )
+                                    : intl.formatMessage(DEFAULT_STORE_LOCATOR_COUNTRY.countryName)
+                            }`
+                            : intl.formatMessage({
+                                  id: 'store_locator.description.viewing_near_your_location',
+                                  defaultMessage: 'Viewing stores near your location'
+                              })}
                     </Box>
                 </AccordionItem>
                 <StoresList storesInfo={storesInfo} />
             </Accordion>
+            {searchStoresParams.limit < numStores &&
+            searchStoresParams.limit < NUM_STORES_PER_REQUEST_API_MAX ? (
+                <Box marginTop="10px">
+                    <Button
+                        key="load-more-button"
+                        onClick={() => {
+                            setSearchStoresParams({
+                                ...searchStoresParams,
+                                limit:
+                                    searchStoresParams.limit + STORE_LOCATOR_NUM_STORES_PER_LOAD <=
+                                    NUM_STORES_PER_REQUEST_API_MAX
+                                        ? searchStoresParams.limit +
+                                          STORE_LOCATOR_NUM_STORES_PER_LOAD
+                                        : searchStoresParams.limit
+                            })
+                        }}
```

Comment:
The way we currently implement the load-more is by incrementing the `limit` 🤔 Each request would grab more and more data, but isn't that a bit wasteful? Because you'll refetch some stores that you've gotten in the previous requests.

Have we considered other alternatives?
- Incrementing the `offset` instead ← optimizing the response size, but more complexity in the UI to piece together all the stores
- Fetching all the stores right from the beginning ← so the Load More button is simply for showing more stores in the UI
- Load More only clickable _once_ ← so the button acts like fetching the rest of the stores